PSTenable/Public/Get-PSTenableSeverity.ps1

```diff
@@ -26,66 +26,93 @@ function Get-PSTenableSeverity {
             'Medium',
             'Low'
         )]
-        [string]
-        $Severity
+        [string[]]
+        $Severity,
+        
+        [Parameter(Position = 0, Mandatory = $false)]
+        [int]$maxrecords = 0
     )
 
     begin {
 
         $TokenExpiry = Invoke-PSTenableTokenStatus
         if ($TokenExpiry -eq $True) {Invoke-PSTenableTokenRenewal}
 
+        $ID = "" 
         switch ($Severity) {
-            "Critical" { $ID = "4" }
-            "High" { $ID = "3" }
-            "Medium" { $ID = "2" }
-            "Low" { $ID = "1" }
+            "Critical" { $ID += "4" }
+            "High" { $ID += "3" }
+            "Medium" { $ID += "2" }
+            "Low" { $ID += "1" }
         }
+        $ID = ($ID -split "" | ? {$_}) -join ","
 
     }
 
     process {
 
-        $query = @{
-            "tool"       = "vulnipdetail"
-            "sortField"  = "cveID"
-            "sortDir"    = "ASC"
-            "type"       = "vuln"
-            "sourceType" = "cumulative"
-            "query"      = @{
-                "name"         = ""
-                "description"  = ""
-                "context"      = ""
-                "status"       = "-1"
-                "createdTime"  = 0
-                "modifiedtime" = 0
-                "sourceType"   = "cumulative"
-                "sortDir"      = "desc"
-                "tool"         = "listvuln"
-                "groups"       = "[]"
-                "type"         = "vuln"
-                "startOffset"  = 0
-                "endOffset"    = 5000
-                "filters"      = [array]@{
-                    "id"           = "severity"
-                    "filterName"   = "severity"
-                    "operator"     = "="
+        $APIresults = @()
+        $CurrentStartOffset = 0
+        
+
+
+        Do {
+            #we're gonna paginate at 5000 rows
+            if ($MaxRecords -ne 0 -and $MaxRecords -lt ($CurrentStartOffset + 4999)) { 
+                $CurrentEndOffset = $MaxRecords
+            } else {
+                $CurrentEndOffset = ($CurrentStartOffset + 4999)
+            }

```

Comment:
prepare and begin tracking Offsets
if the maxrecord is between the current start and 5000 above start, set the endOffset to the maxrecord